### PR TITLE
Merge auth_crypto into base

### PR DESCRIPTION
**Pull request by xmo-odoo**
https://www.odoo.com/web#id=34211&view_type=form&model=project.task&action=333&active_id=133&menu_id=4720

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Is there the possibility of keeping the translations already done of the old module?

**Comment by pedrobaeza** [comment]
You're right. I though the field encrypted was preserved, but indeed there's no sense for that.

Thanks.

File: odoo/addons/base/models/res_users.py [hunk 1]
```diff
@@ -233,11 +247,59 @@ def _get_company(self):
...
+    def init(self):
+        cr = self.env.cr
+
+        # allow setting plaintext passwords via SQL and have them
+        # automatically encrypted at startup: look for passwords which don't
+        # match the "extended" MCF and pass those through passlib.
+        # Alternative: iterate on *all* passwords and use CryptContext.identify
+        cr.execute("""
+        SELECT id, password FROM res_users
+        WHERE active
```
Comment (KangOl) [inline odoo/addons/base/models/res_users.py]: Why only active users?

File: odoo/addons/base/models/res_users.py [hunk 2]
```diff
@@ -19,7 +20,18 @@
 _logger = logging.getLogger(__name__)
 
 # Only users who can modify the user (incl. the user herself) see the real contents of these fields
-USER_PRIVATE_FIELDS = ['password']
+USER_PRIVATE_FIELDS = []
+
+DEFAULT_CRYPT_CONTEXT = passlib.context.CryptContext(
+    # kdf which can be verified by the context. The default encryption kdf is
+    # the first of the list
+    ['pbkdf2_sha512', 'plaintext'],
```
Comment (KangOl) [inline odoo/addons/base/models/res_users.py]: why not keep `md5_crypt`?

Comment (xmo-odoo) [reply odoo/addons/base/models/res_users.py]: Because it was deprecated 4 years ago, and I'm not sure auth_crypt was even auto-installed at the time.

**Pull request merged**